.github/workflows/coverage.yaml

```diff
@@ -149,13 +151,32 @@ jobs:
 
           # fill new tests
           # using `|| true` here because if no tests found, pyspec fill returns error code
+          FOUND_TEST=false
           mkdir -p fixtures/state_tests
           mkdir -p fixtures/eof_tests
-          echo "$files" | while read line; do
+
+          # Use a while loop with a here-string to avoid subshell issues
+          while IFS= read -r line; do
             file=$(echo "$line" | cut -c 3-)
-            fill $file --until=Cancun --evm-bin evmone-t8n || true >> filloutput.log 2>&1
-            (fill $file --fork=CancunEIP7692 --evm-bin evmone-t8n -k eof_test || true) > >(tee -a filloutput.log filloutputEOF.log) 2>&1
-          done
+            if grep -q "def test_" "$file"; then
```

Comment:
This is kinda risky, pytest also loads `class Test*` as tests, but even after adding that option too, it seems fragile still.

I looked up for an alternative and there is this pytest plugin that fills only modified tests: https://github.com/anapaulagomes/pytest-picked

I tried it out with `fill` and it seems to work out of the box. We could add it to the setup (`pytest-picked==0.5.0` in the `install_requires` of `setup.cfg`), use `fill --picked` to fill the modified tests and then maybe checkout the base branch and then do `fill --picked --mode=branch --parent-branch=origin/PR-${{ github.event.pull_request.number }}` to fill the original tests.

Reply:
If you do that it will be same as before.
Where I just pass each .py to fill and it fills if there are tests. 

This condition here is only to skip the script if there are no actual tests introduced by the pr. Because previously it would error that no tests generated.


 Which is also unlikely situation as a pr that touchs tests folder will most often touch the actual test files.

So in short I think this logic could be rolled back and removed if you think its fragile.

Reply:
I have disables test detection of def test_  this is also safer so we won't skip the coverage script. 
new tests over no tests ==> green coverage

Reply:
also fill --picked could is nice, but I am already building a file array to print it in the script for debug information. I could just iterate over it. I make this array global now